docs/proposals/autoscaling.md

```diff
@@ -0,0 +1,245 @@
+
+The current, most obvious, use case is scaling an application based on network traffic like requests per second.  Most 
+applications will expose one or more network endpoints for clients to connect to. Many of those endpoints will be load 
+balanced or situated behind a proxy - the data from those proxies and load balancers can be used to estimate client to 
+server traffic for applications. This is the primary, but not sole, source of data for making decisions.
+
+Within Kubernetes a [kube proxy](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services.md#ips-and-portals) 
+running on each node directs service requests to the underlying implementation.  
+
+While the proxy provides internal inter-pod connections, there will be L3 and L7 proxies and load balancers that manage 
+traffic to backends. OpenShift, for instance, adds a "route" resource for defining external to internal traffic flow. 
+The "routers" are HAProxy or Apache load balancers that aggregate many different services and pods and can serve as a 
+data source for the number of backends.
+
+### Scaling based on predictive analysis
+
+Scaling may also occur based on predictions of system state like anticipated load, historical data, etc.  Hand in hand 
+with scaling based on traffic, predictive analysis may be used to determine anticipated system load and scale the application automatically.  
+
+### Scaling based on arbitrary data
+
+Administrators may wish to scale the application based on any number of arbitrary data points such as job execution time or
+duration of active sessions.  There are any number of reasons an administrator may wish to increase or decrease capacity which 
+means the auto-scaler must be a configurable, extensible component.
+
+## Specification
+
+In order to facilitate talking about auto-scaling the following definitions are used:
+
+* `ReplicationController` - the first building block of auto scaling.  Pods are deployed and scaled by a `ReplicationController`.
+* kube proxy - The proxy handles internal inter-pod traffic, an example of a data source to drive an auto-scaler
+* L3/L7 proxies - A routing layer handling outside to inside traffic requests, an example of a data source to drive an auto-scaler
+* auto-scaler - scales replicas up and down by using the `resize` endpoint provided by scalable resources (`ReplicationController`)
+
+
+### Auto-Scaler
+
+The Auto-Scaler is a state reconciler responsible for checking data against configured scaling thresholds 
+and calling the `resize` endpoint to change the number of replicas.  The scaler will 
+use a client/cache implementation to receive watch data from the data aggregators and respond to them by 
+scaling the application.  Auto-scalers are created and defined like other resources via REST endpoints and belong to the 
+namespace just as a `ReplicationController` or `Service`.
+
+Since an auto-scaler is a durable object it is best represented as a resource.  
+
+```go
```

Comment:
A lot of the code lines overflow off the right side of the text box when I view the .md file, with no horizontal scrollbar at the bottom (so it's just not possible to see the overflow text). If this is how it will look when it is viewed after being merged, I'd suggest breaking up these lines. The way I am viewing the .md file is go to "files changed" in the PR and click "View"